### PR TITLE
Convert Traits module into ts

### PR DESCRIPTION
**Pull request by xQwexx**
I am planning to do a bigger refactor overhaul of the traits just first I want to have as typescript

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -516,7 +516,7 @@ declare namespace grapesjs {
 
   interface TraitView {
     noLabel?: boolean;
-    templateInput?: string | ((options: { trait: Trait }) => string);
+    templateInput?: string;
```
Comment (artf) [inline index.d.ts:519]: I see what you have done in View classes but we can't remove the support of `templateInput` as a function as it's necessary for the current customization mechanism and it is also publicly documented.

Comment (artf) [inline index.d.ts:519]: We can't remove the support of `templateInput` as a function as it's necessary for the current customization mechanism.

File: src/trait_manager/view/TraitView.ts [hunk 2]
```diff
@@ -92,42 +112,27 @@ export default class TraitView extends Backbone.View {
...
-      tpl =
-        this.createLabel({
-          label,
-          component: target,
-          trait: this,
-        }) || '';
-    }
+  private renderLabel() {
+    const { $el } = this;
+    let tpl = this.templateLabel;
```
Comment (artf) [inline src/trait_manager/view/TraitView.ts:135]: I know it doesn't make sense from the class perspective to have undefined methods but we can't simply remove methods publicly documented for customization like `createLabel`
https://grapesjs.com/docs/modules/Traits.html#update-layout

Comment (artf) [inline src/trait_manager/view/TraitView.ts:135]: I know it doesn't make sense from the class perspective to have undefined methods but we can't simply remove methods publicly documented for customization like `createLabel`
https://grapesjs.com/docs/modules/Traits.html#update-layout

**Review by artf — CHANGES_REQUESTED** [review]

File: index.d.ts (hunk 1 above)
Comment (xQwexx) [reply index.d.ts]: I added it back

File: src/trait_manager/view/TraitView.ts (hunk 2 above)
Comment (xQwexx) [reply src/trait_manager/view/TraitView.ts:137]: I reverted it but I think these are redundant capabilities so I would like to get ride of them sometimes

File: src/trait_manager/index.ts [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+        config,
+      },
+      this.getTypes()
+    );
+    this.view = view;
+    return view.el;
+  }
+
+  destroy() {
+    //this.__destroy();
```
[the remark this answers is not in the document]
Comment (xQwexx) [reply src/trait_manager/index.ts]: Yes it was intentional I also deleted the destroy function

**Comment by artf** [comment]
Thanks @xQwexx can you please resolve pending conflicts so I'll be able to review it again?

**Comment by artf** [comment]
I'm closing this one as traits module is now converted to ts

**Pull request closed without merging**